BrowserKit/Sources/SiteImageView/ImageProcessing/BundleImageFetcher/BundleDataProvider.swift

```diff
@@ -0,0 +1,50 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0
+
+import UIKit
+
+protocol BundleDataProvider {
+    func getBundleData() throws -> Data
+
+    func getPath(from path: String) -> String?
+
+    func getBundleImage(from path: String) -> UIImage?
+}
+
+class DefaultBundleDataProvider: BundleDataProvider {
+
+    enum BundleError: Error {
+        case noPath
+    }
+
+    func getBundleData() throws -> Data {
+        guard let path = bundle.path(forResource: "top_sites", ofType: "json") else {
```

Comment:
Is this only for TopSite? Will we use this in the future for any other type?

Reply:
Top sites is the primary use case it exists but in theory any favicon from a domain that has a bundled favicon should get it from here first to avoid doing the work of getting it from the web. We can probably rename the file to something more generic once everything that relies on it in the main app is migrated.

Reply:
Noted this down in the clean up task https://mozilla-hub.atlassian.net/browse/FXIOS-5296